### PR TITLE
Upgrade to Avro 0.10.1

### PR DESCRIPTION
**Pull request by vlk32**
Hello David,

I'd like to upgrade plugin to Avro release 0.10.1, which contains

*  one improvement for support `local_timestamp_ms -> LocalDateTime` 
   * see: https://issues.apache.org/jira/browse/AVRO-2924
* and fix for one significant bug 
   * see: https://issues.apache.org/jira/browse/AVRO-2702

I've prepared pull-request with unit-test for AVRO-2924.

thak's for accepting it

vlk

### CODE REVIEW

**Comment by davidmc24** [comment]
Thanks for the pull request. I'll try to take a look tonight.

**Comment by davidmc24** [comment]
Pull request looks pretty good.  My understanding, from looking at your changes and https://issues.apache.org/jira/browse/AVRO-2924, is that the main relevant change is that when using `SpecificCompiler` in 1.10.1, it now handles a couple cases as `java.time.LocalDateTime` instead of `long`.  This appears to be a compile-time behavior change that is runtime-compatible with 1.10.0.

I'll be suggesting a couple minor changes to indicate this runtime compatibility and related testing.

File: README.md [hunk 1]
```diff
@@ -19,8 +19,9 @@ This is a [Gradle](http://www.gradle.org/) plugin to allow easily performing Jav
     * If you need support for Gradle 4.4-5.0, version 0.18.0 was the last version tested for compatibility
     * If you need support for Gradle 3.0-3.5.1 or 4.0-4.3, version 0.17.0 was the last version tested for compatibility
     * If you need support for Gradle 2.0-2.14.1, version 0.9.1 was the last version tested for compatibility; please see [the Gradle plugin portal](https://plugins.gradle.org/plugin/com.commercehub.gradle.plugin.avro)
-* Currently built against Avro 1.10.0
-    * Currently tested against Avro 1.10.0
+* Currently built against Avro 1.10.1
+    * Currently tested against Avro 1.10.1
+    * If you need support for Avro 1.10.0 try plugin version 0.20.0
```
Comment (davidmc24) [inline README.md:24]:
```suggestion
    * Currently tested against Avro 1.10.0-1.10.1
```

File: build.gradle [hunk 2]
```diff
@@ -14,7 +14,7 @@ repositories {
     jcenter()
 }
 
-def compileAvroVersion = "1.10.0"
+def compileAvroVersion = "1.10.1"
```
Comment (davidmc24) [inline build.gradle:17]: Please add "1.10.1" to `def avroVersions` on line 210, to support runtime compatibility testing of both versions.

File: src/test/groovy/com/commercehub/gradle/plugin/avro/AvroPluginFunctionalSpec.groovy [hunk 3]
```diff
@@ -59,16 +59,24 @@ class AvroPluginFunctionalSpec extends FunctionalSpec {
...
         result.task(":generateAvroProtocol").outcome == SUCCESS
         result.task(":generateAvroJava").outcome == SUCCESS
         result.task(":compileJava").outcome == SUCCESS
+        projectFile("build/generated-main-avro-java/org/apache/avro/Interop.java").file
         projectFile(buildOutputClassPath("org/apache/avro/Foo.class")).file
         projectFile(buildOutputClassPath("org/apache/avro/Interop.class")).file
         projectFile(buildOutputClassPath("org/apache/avro/Kind.class")).file
         projectFile(buildOutputClassPath("org/apache/avro/MD5.class")).file
         projectFile(buildOutputClassPath("org/apache/avro/Node.class")).file
+        interopJavaContent
```
Comment (davidmc24) [inline src/test/groovy/com/commercehub/gradle/plugin/avro/AvroPluginFunctionalSpec.groovy:74]: This is a good addition.  Thank you for suggesting it.  Do you think there's value in also adding assertions for the more "normal" fields, or would that be overkill?

File: src/test/resources/com/commercehub/gradle/plugin/avro/interop.avdl [hunk 4]
```diff
@@ -45,6 +45,11 @@ protocol InteropProtocol {
     Kind enumField;
     MD5 fixedField;
     Node recordField;
+    decimal(10,2) decimalField;
+    date dateField;
+    time_ms timeField;
+    timestamp_ms timeStampField;
+    local_timestamp_ms localTimeStampField;
```
Comment (davidmc24) [inline src/test/resources/com/commercehub/gradle/plugin/avro/interop.avdl:52]: Are there any other types that we might want coverage for?
It looks to me like we already have most of what is supported [here](https://github.com/apache/avro/blob/7d1e63b219e6d0778bc57195152477adee97fcab/lang/java/compiler/src/main/javacc/org/apache/avro/compiler/idl/idl.jj#L215).  Maybe `uuid`?

**Review by davidmc24 — CHANGES_REQUESTED** [review]
Could you please update `avroVersion` in `test-project/build.gradle`?

File: README.md (hunk 1 above)
Comment (vlk32) [reply README.md]: O.K. Fixed.

File: src/test/groovy/com/commercehub/gradle/plugin/avro/AvroPluginFunctionalSpec.groovy (hunk 3 above)
Comment (vlk32) [reply src/test/groovy/com/commercehub/gradle/plugin/avro/AvroPluginFunctionalSpec.groovy:75]: Done. For all datatypes ( which I know ).

File: src/test/resources/com/commercehub/gradle/plugin/avro/interop.avdl (hunk 4 above)
Comment (vlk32) [reply src/test/resources/com/commercehub/gradle/plugin/avro/interop.avdl:52]: I  think, that there is not any.
And `uuid` is not compilable with Avro 1.10.1.

**Comment by davidmc24** [comment]
Merged.  Thanks again for the pull request.  I'll release a version including it no later than the end of the week.

**Pull request merged**